### PR TITLE
Add Notebooks API Support

### PR DESCRIPTION
**Pull request by gogasca**
**Release Note Template for Downstream PRs (will be copied)**

```release-note:new-resource
`google_notebook_instance`
```
```release-note:new-resource
`google_notebook_environment`
```

### CODE REVIEW

**Comment by ndmckinley** [comment]
This looks great to me!  Hey!  :)

Have you been able to run the tests?  The only thing I'd ask is a test for Environment, other than that this seems great.

File: third_party/terraform/tests/resource_notebooks_instance_test.go.erb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+			},
+		},
+	})
+}
+
+func testAccNotebooksInstance_update(prefix, name string) string {
+	return fmt.Sprintf(`
+
+resource "google_notebooks_instance" "test" {
+  name = "%s"
```
Comment (ndmckinley) [inline third_party/terraform/tests/resource_notebooks_instance_test.go.erb:47]: This call has only one `%s` but accepts two arguments - this upsets the linter, so if you can fix it it'll be a little nicer.  :)

**Comment by gogasca** [comment]
> This looks great to me! Hey! :)
> 
> Have you been able to run the tests? The only thing I'd ask is a test for Environment, other than that this seems great.

Thanks for the fast response, I still some guidance in how to run those tests. Thanks

File: third_party/terraform/tests/resource_notebooks_instance_test.go.erb (hunk 1 above)
Comment (gogasca) [reply third_party/terraform/tests/resource_notebooks_instance_test.go.erb:47]: ack, I fixed, but Build still failing, (try to get access to logs but I don't have permission) any pointers will be great thanks

**Comment by ndmckinley** [comment]
It looks like there's an issue with the tests.

```
TestAccNotebooksInstance_update: testing.go:674: Step 0 error: config is invalid: 2 problems:
- Missing required argument: The argument "machine_type" is required, but no definition was found.
- Unsupported argument: An argument named "node_count" is not expected here.
--- FAIL: TestAccNotebooksInstance_update (0.17s)
```

**Comment by gogasca** [comment]
In acceptance test I get: (TestAccNotebooksInstance_create)

```
------- Stdout: -------
=== RUN   TestAccNotebooksInstance_create
=== PAUSE TestAccNotebooksInstance_create
=== CONT  TestAccNotebooksInstance_create
TestAccNotebooksInstance_create: testing.go:674: Step 0 error: After applying this step, the plan was not empty:
DIFF:
```
Any idea why?

**Pull request merged**